extension/src/popup/helpers/searchAsset.ts

```diff
@@ -86,39 +58,77 @@ export interface TokenRecord {
   decimals: number;
 }
 
+export type VerifiedTokenRecord = TokenRecord & { verifiedLists: string[] };
+
 export const getVerifiedTokens = async ({
   networkDetails,
   contractId,
   setIsSearching,
+  assetsLists,
 }: {
   networkDetails: NetworkDetails;
   contractId: string;
   setIsSearching?: (isSearching: boolean) => void;
+  assetsLists: AssetsLists;
 }) => {
-  let verifiedTokens = [] as TokenRecord[];
-
-  const fetchVerifiedTokens = async () => {
-    const verifiedTokenRes = await searchToken({
-      networkDetails,
-      onError: (e) => {
-        console.error(e);
-        if (setIsSearching) {
-          setIsSearching(false);
-          captureException("Unable to search stellar.expert token list");
-        }
-      },
-    });
+  const networkLists = assetsLists[networkDetails.network as AssetsListKey];
+  const promiseArr = [];
+  const nativeContract = getNativeContractDetails(networkDetails);
+
+  if (contractId === nativeContract.contract) {
+    return [{ ...nativeContract, verifiedLists: [] }];
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const networkList of networkLists) {
+    const { url = "" } = networkList;
+
+    const fetchAndParse = async () => {
+      let res;
+      try {
+        res = await fetch(url);
+      } catch (e) {
+        captureException(`Failed to load asset list: ${url}`);
+      }
+
+      return res?.json();
+    };
+
+    promiseArr.push(fetchAndParse());
+  }
+
+  const promiseRes = await Promise.allSettled(promiseArr);
+
+  const verifiedTokens = [] as VerifiedTokenRecord[];
+
+  let verifiedToken = {} as TokenRecord;
+  const verifiedLists: string[] = [];
 
-    verifiedTokens = verifiedTokenRes.filter((record: TokenRecord) => {
-      const regex = new RegExp(contractId, "i");
-      if (record.contract.match(regex)) {
-        return true;
+  promiseRes.forEach((r) => {
+    if (r.status === "fulfilled") {
+      const list = r.value?.tokens ? r.value?.tokens : r.value?.assets;
```

Comment:
in the future, these should all adhere to the SEP0042 spec where the key will be called `tokens`. Adding this check for now while I wait for Soroswap to fix this on their end